pages/profile/[username].tsx

```diff
@@ -107,11 +110,20 @@ const UserProfile: React.FC = () => {
       },
       {}
     )
+
     let starsReceived = [] as Star[]
     if (lessonStatusMap[String(lesson.id)]) {
       starsReceived = lessonStatusMap[String(lesson.id)].starsReceived as Star[]
     }
 
+    lessonStatus.forEach(({ starsReceived }) => {
```

Comment:
You are using the profileStars variable and the forEach to do a combination of reducing and filtering. You should separate out this logic, and use filter along with a more appropriate higher order function so that your code becomes more explicit and clear about what is happening. It can be made even more simple by removing the nested forEach inside with a spread operator instead.

const validProfiles = lessonStatus.filter(({ starsReceived }) => starsReceived)
const profileStars: Star[] = validProfiles.reduce((acc, { starsReceived }) => {
  acc.push(...starsReceived)
  return acc
}, [])